src/rockset_sqlalchemy/client/cursor.py

```diff
@@ -112,20 +113,36 @@ def fetchmany(self, size=None):
                 break
             docs.append(doc)
         return docs
+        
 
     @property
     def description(self):
+        # if no query has been executed, return None
         if self._response is None:
             return None
 
-        desc = []
-        for field in self._response_to_column_fields(self._response.column_fields):
-            name, type_ = field["name"], field["type"]
-            null_ok = name != "_id" and "__id" not in name
-
-            # name, type_code, display_size, internal_size, precision, scale, null_ok
-            desc.append((name, type_, None, None, None, None, null_ok))
-        return desc
+        if hasattr(self._response, "column_fields"): # the query was a DESCRIBE call
```

Comment:
column_fields would be present for SELECT queries as well. It wouldn't be in the response iff the is doing a `SELECT *`.  Did you test this with something like `SELECT a, b from foo`?